pkg/adapter/mtping/runner.go

```diff
@@ -122,9 +125,21 @@ func (a *cronJobsRunner) Stop() {
 
 func (a *cronJobsRunner) cronTick(ctx context.Context, event cloudevents.Event) func() {
 	return func() {
+
+		event := event.Clone()
+		event.SetID(uuid.New().String())
+		target := cecontext.TargetFrom(ctx).String()
+		source := event.Context.GetSource()
+		time.Sleep(time.Duration(rand.Intn(500)) * time.Millisecond) // provide a delay so not all ping fired instantaneously distribute load on resources.
+
+		if a.Logger.Desugar().Core().Enabled(zap.DebugLevel) {
+			a.Logger.Debugf("sending cloudevent id %s, source: %s, target: %s", event.ID(), source, target)
+		}
+
 		if result := a.Client.Send(ctx, event); !cloudevents.IsACK(result) {
 			// Exhausted number of retries. Event is lost.
-			a.Logger.Error("failed to send cloudevent", zap.Any("result", result))
+			a.Logger.Error("failed to send cloudevent result:", zap.Any("result", result),
+				", source: ", source, ", target: ", target, ", id: ", event.ID())
```

Comment:
Can you use zap.String instead of `", source: ", source, "`, target: ", target, etc...?

Reply:
Sure.

Reply:
Changed